pkg/model/dns.go

```diff
@@ -89,20 +95,10 @@ func (b *DNSModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		// Pulling this down into it's own if statement. The DNS configuration here
```

Comment:
I think some of the logic here could be better, looks like the if statement I pulled down could probably be combined how you had put it in the first if statement..

But either way.. the logic is sound - we can boyscout / cleanup later